pkg/core/xds/rules.go

```diff
@@ -139,6 +138,7 @@ func BuildRules(list []PolicyItem) (Rules, error) {
 		}
 		// 3. For each combination determine a configuration
 		confs := []any{}
+		distinctOrigins := map[core_model.ResourceMeta]struct{}{}
```

Comment:
`core_model.ResourceMeta` interface can contain a pointer as a Value, I don't think we can use it as a key

Reply:
Yeah, I was on the fence, but the meta should be exactly the same even if there is pointer, no? Let me change it to ResourceKey just to be sure...